src/ckeditor.jsx

```diff
@@ -51,56 +71,83 @@ export default class CKEditor extends React.Component {
 	}
 
 	_initializeEditor() {
-		this.props.editor
-			.create( this.domContainer.current, this._getConfig() )
-			.then( editor => {
-				this.editor = editor;
-
-				if ( 'disabled' in this.props ) {
-					editor.isReadOnly = this.props.disabled;
-				}
-
-				if ( this.props.onInit ) {
-					this.props.onInit( editor );
-				}
-
-				const modelDocument = editor.model.document;
-				const viewDocument = editor.editing.view.document;
-
-				modelDocument.on( 'change:data', event => {
-					/* istanbul ignore else */
-					if ( this.props.onChange ) {
-						this.props.onChange( event, editor );
+		const creator = ( el, config ) => {
+			return this.props.editor.create( el, config )
+				.then( editor => {
+					if ( 'disabled' in this.props ) {
+						editor.isReadOnly = this.props.disabled;
 					}
-				} );
 
-				viewDocument.on( 'focus', event => {
-					/* istanbul ignore else */
-					if ( this.props.onFocus ) {
-						this.props.onFocus( event, editor );
-					}
+					const modelDocument = editor.model.document;
+					const viewDocument = editor.editing.view.document;
+
+					modelDocument.on( 'change:data', event => {
+						/* istanbul ignore else */
+						if ( this.props.onChange ) {
+							this.props.onChange( event, editor );
+						}
+					} );
+
+					viewDocument.on( 'focus', event => {
+						/* istanbul ignore else */
+						if ( this.props.onFocus ) {
+							this.props.onFocus( event, editor );
+						}
+					} );
+
+					viewDocument.on( 'blur', event => {
+						/* istanbul ignore else */
+						if ( this.props.onBlur ) {
+							this.props.onBlur( event, editor );
+						}
+					} );
+
+					// The onInit should be fired once the `editor` property
+					// can be reached from the <ckeditor> component.
+					// Ideally this part should be moved to the watchdog item creator listeners.
+					setTimeout( () => {
+						if ( this.props.onInit ) {
+							this.props.onInit( this.editor );
+						}
```

Comment:
The `timeout` solution is just bad.
Let's move the callback execution after creating an editor by Watchdog.

```js
this.watchdog.create( this.domContainer.current, this._getConfig() )
  .then( /* the code goes here */ ) 
  .catch( onError );
```

Reply:
We agreed to change `onInit` to `onReady`.